cpp/src/parquet/arrow/reader_internal.h

```diff
@@ -114,11 +115,12 @@ struct ReaderContext {
   ::arrow::MemoryPool* pool;
   FileColumnIteratorFactory iterator_factory;
   bool filter_leaves;
-  std::unordered_set<int> included_leaves;
+  std::vector<int> included_leaves;
 
   bool IncludesLeaf(int leaf_index) const {
     return (!this->filter_leaves ||
-            (included_leaves.find(leaf_index) != included_leaves.end()));
```

Comment:
Why not keep the `unordered_set`? Does the `vector` make things faster?
(or you may want to use a `vector<bool>` indexed by node index)

Reply:
On the subject, can you also elide `filter_leaves` and use include_leaves.empty() instead?

Reply:
Reverted to unordered_set. Leaving the filter_leaves thing unchanged since there is other business logic